cmd/plume/prerelease.go

```diff
@@ -388,7 +506,19 @@ func awsUploadToPartition(spec *channelSpec, part *awsPartitionSpec, imageName,
 	}
 	defer f.Close()
 
-	s3ObjectPath := fmt.Sprintf("%s/%s/%s", specBoard, specVersion, strings.TrimSuffix(spec.AWS.Image, filepath.Ext(spec.AWS.Image)))
+	awsImageMetadata, err := getSpecAWSImageMetadata(spec)
+	if err != nil {
+		return nil, nil, fmt.Errorf("Could not generate the image metadata: %v", err)
+	}
+
+	imageFileName := awsImageMetadata["imageFileName"]
+	imageName = awsImageMetadata["imageName"]
+	imageDescription = awsImageMetadata["imageDescription"]
+
+	s3ObjectPath := fmt.Sprintf("%s/%s/%s", specBoard, specVersion, strings.TrimSuffix(imageFileName, filepath.Ext(imageFileName)))
+	if selectedDistro == "fedora" {
```

Comment:
This `if` block seems redundant. It's setting `s3ObjectPath` to the same value as it was initially instantiated to above.